src/other/utils.jl

```diff
@@ -181,6 +181,34 @@ end
 
 _findall(B) = findall(B)
 
+function _findall(B::AbstractVector{Bool})
+    @assert firstindex(B) == 1
+    nnzB = count(B)
+
+    # fast path returning range
+    nnzB == 0 && return 1:0
+    len = length(B)
+    nnzB == len && return 1:len
+    start::Int = findfirst(B)
+    nnzB == 1 && return start:start
+    start + nnzB - 1 == len && return start:len
+    stop::Int = findnext(!, B, start + 1) - 1
+    start + nnzB == stop + 1 && return start:stop
+
+    # slow path returning Vector{Int}
+    I = Vector{Int}(undef, nnzB)
+    I[1:stop - start + 1] .= start:stop
```

Comment:
I will wait with merging till @mbauman comments on https://github.com/JuliaData/DataFrames.jl/pull/2771 about the allocations induced by broadcasting in such scenarios.